eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/protocol/http/processor/UnSubscribeProcessor.java

```diff
@@ -131,12 +126,15 @@ public void onResponse(HttpCommand httpCommand) {
 
         synchronized (eventMeshHTTPServer.localClientInfoMapping) {
             boolean isChange = true;
+
+            registerClient(unSubscribeRequestHeader, consumerGroup, unSubTopicList, unSubscribeUrl);
+
             for (String unSubTopic : unSubTopicList) {
                 List<Client> groupTopicClients = eventMeshHTTPServer.localClientInfoMapping.get(consumerGroup + "@" + unSubTopic);
                 Iterator<Client> clientIterator = groupTopicClients.iterator();
                 while (clientIterator.hasNext()) {
                     Client client = clientIterator.next();
-                    if (StringUtils.equals(client.ip, ip)) {
+                    if (StringUtils.equals(client.ip, ip) && StringUtils.equals(client.url, unSubscribeUrl)) {
```

Comment:
**client.ip** here should change to use **client.pid** as for equals condition，here ip represent ip+port and the port is random this equal condition will never reached, and client won't unsubscribe

Reply:
Thanks @xwm1992 I fixed it by using client.pid instead. Thanks.